tests/pyccel/test_pyccel.py

```diff
@@ -20,19 +29,40 @@ def get_fortran_output(path_dir,test_file):
     assert(p.returncode==0)
     return out
 
-def pyccel_test(test_file):
+def pyccel_test(test_file, dependencies = None):
     base_dir = os.path.dirname(os.path.realpath(__file__))
     path_dir = os.path.join(base_dir, 'scripts')
 
     pyth_out = get_python_output(path_dir,test_file)
+    if (isinstance(dependencies, list)):
+        for d in dependencies:
+            compile_pyccel(path_dir, d)
+    elif (isinstance(dependencies, str)):
+        compile_pyccel(path_dir, dependencies)
     compile_pyccel(path_dir,test_file)
     fort_out = get_fortran_output(path_dir,test_file[:-3])
 
     assert(pyth_out.strip()==fort_out.strip())
 
 @pytest.mark.xfail
 def test_imports():
-    pyccel_test("test_imports.py")
+    pyccel_test("test_imports.py","funcs.py")
 
 def test_funcs():
     pyccel_test("test_funcs.py")
+
+def test_f2py_compat():
```

Comment:
@EmilyBourne We should remove any `.so` module from the `path_dir` directory before running this test, otherwise Python will try to import `return_one` from the shared library rather then the pure Python one (possibly raising an `ImportError`). You can achieve this by adding a `teardown_module` function to this file, where you remove the shared library just created.

Reply:
Instead of calling a `teardown` function explicitly, I would prefer to create a `teardown_module` function that gets called automatically by `pytest` (see https://docs.pytest.org/en/latest/xunit_setup.html). A `setup_module` function also makes sense in this case.

Both functions (`setup_module` and `teardown_module`) could call your own `teardown` function. The advantage of using this structure is that `pytest` takes care of calling these functions even in the case of an assertion error.

Reply:
I noticed that the function teardown was being called anyway after every function. I think this is better in case multiple tests want to use the same file. E.g once to test functions then once to test imports by importing the file containing the functions. I haven't seen this functionality documented though, only teardown_module, teardown_function(function), teardown_class